EIPS/eip-7928.md

```diff
@@ -79,76 +81,81 @@ class AccountBalanceDiff(Container):
 
 BalanceDiffs = List[AccountBalanceDiff, MAX_ACCOUNTS]
 
-# Post-transaction nonce structures
+# Code Diff structures
+class CodeChange(Container):
+    tx_index: TxIndex
+    new_code: CodeData
+
+class AccountCodeDiff(Container):
+    address: Address
+    changes: List[CodeChange, MAX_TXS]
+
+CodeDiffs = List[AccountCodeDiff, MAX_ACCOUNTS]
+
+# Pre-transaction nonce structures
 class AccountNonce(Container):
     address: Address  # account address
-    nonce_before: Nonce  # nonce value after block execution
+    nonce_before: Nonce  # nonce value before block execution
```

Comment:
I'm confused by this one. Why do we need this value in the BAL? If it is the "nonce value before block execution" then we can read this from disk (just like how we treat storage values, first read -> read from disk)

Reply:
I meant "before transaction execution here", So, we need the pre-tx nonce to know where to deploy contracts where the deployer is a contract using CREATE or CREATE2.